### PR TITLE
stac-n baremetal playbook and roles

### PR DESCRIPTION
**Pull request by zvonkok**
Added stac-n baremetal playbook and roles to install openonload discover-hw-topology and
create a stac-n tuned profile according to the discovered features.

### CODE REVIEW

File: playbooks/roles/solarflare_discover_hw_topology/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  shell: |
+    IFS=',' read -r -a  array <<< `cat "{{ out_file_isolcpus }}"`
+    echo "${array[0]}"
+  register: local_housekeeping_core_0
+
+- name: register second local housekeeping core
+  shell: |
+    IFS=',' read -r -a  array <<< `cat "{{ out_file_isolcpus }}"`
+    echo "${array[1]}"
+  register: local_housekeeping_core_1
```
Comment (jeremyeder) [inline playbooks/roles/solarflare_discover_hw_topology/tasks/main.yml]: Does this imply that we must have more than 1 housekeeping core?

File: playbooks/roles/solarflare_install_onload/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
+---
+svt_repo:       github.com/openshift/svt-private.git
```
Comment (jeremyeder) [inline playbooks/roles/solarflare_install_onload/defaults/main.yml]: Is there any possibility to refactor this to pull from the vendor download page:

http://www.openonload.org/download/

File: playbooks/roles/solarflare_install_onload/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+openonload_tgz: openonload-201805.tgz
+sfutils_rpm:    sfutils-7.1.0.1003-1.x86_64.rpm
+
+onload_srpm: openonload-201805-1.el7.src.rpm
+onload_spec: openonload.spec
+onload_rpms:
+       - openonload-201805-1.el7.x86_64.rpm
+       - openonload-kmod-3.10.0-862.el7-201805-1.el7.x86_64.rpm
+
+solarflare_pkg_prefix: /root/svt-private/networking/synthetic/solarflare/pkg/201805
```
Comment (jeremyeder) [inline playbooks/roles/solarflare_install_onload/defaults/main.yml]: Same here

File: playbooks/roles/stac_n_tuned_configuration/templates/script.sh [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+    echo 400 > /sys/devices/system/node/node{{ tuned_local_numa_node  }}/hugepages/hugepages-2048kB/nr_hugepages
+
+    ifup {{ tuned_test_interface }}
+    ip link set {{ tuned_test_interface }} mtu 1700
+
+    sfcirqaffinity {{ tuned_test_interface }} {{ tuned_local_housekeeping_core_0 }}
+    #sfcirqaffinity p258p2 8
+
+    taskset -pc {{ tuned_local_housekeeping_core_1 }}  `ps -eo pid,comm | grep onload_cp | awk '{ print $1 }'`
```
Comment (jeremyeder) [inline playbooks/roles/stac_n_tuned_configuration/templates/script.sh]: an optimization could be to use the tuned scheduler plugin to handle this.

File: playbooks/roles/solarflare_discover_hw_topology/tasks/main.yml (hunk 1 above)
Comment (zvonkok) [reply playbooks/roles/solarflare_discover_hw_topology/tasks/main.yml]: Yes, we need one for the IRQ, if onload is polling you still get some interrupts not so many but still some arriving,  and the other one for the control plane process.

File: playbooks/roles/solarflare_install_onload/defaults/main.yml (hunk 2 above)
Comment (zvonkok) [reply playbooks/roles/solarflare_install_onload/defaults/main.yml]: Hmm, there are parts I can download from the URL mentioned above, have to check for the other packages.

File: playbooks/roles/solarflare_install_onload/defaults/main.yml (hunk 3 above)
Comment (zvonkok) [reply playbooks/roles/solarflare_install_onload/defaults/main.yml]: See comment above, the onload package would be no problem, have to check the other.

File: playbooks/roles/stac_n_tuned_configuration/templates/script.sh (hunk 4 above)
Comment (zvonkok) [reply playbooks/roles/stac_n_tuned_configuration/templates/script.sh]: Done, tested and verified, the onload_cp_serve is scheduled to the second local housekeeping core.

File: playbooks/roles/solarflare_install_onload/defaults/main.yml (hunk 2 above)
Comment (zvonkok) [reply playbooks/roles/solarflare_install_onload/defaults/main.yml]: Ok all packages coming now from openonload.org/download and channel.solarflare.com, no svt-private needed.

Comment (jeremyeder) [reply playbooks/roles/solarflare_install_onload/defaults/main.yml]: Thank you!

**Comment by jeremyeder** [comment]
Seems like there are still references to svt-private in this PR.  After those are stripped, we can merge.

**Comment by zvonkok** [comment]
Removed any reference to svt. Repo is now clean.

**Pull request merged**